### PR TITLE
feat: allow for configurable source diretory location and selector pr…

### PR DESCRIPTION
**Pull request by Brocco**
…efix at app creation

### CODE REVIEW

File: addon/ng2/blueprints/component/index.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ module.exports = {
             dir = dirParts.join(path.sep);
           }
         }
-        this.appDir = dir.replace(`src${path.sep}client${path.sep}`, '');
+        this.appDir = dir.replace(this.project.ngConfig.defaults.sourceDir + path.sep, '');
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: could you `path.normalize` this?

File: addon/ng2/blueprints/component/index.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ module.exports = {
 
   afterInstall: function(options) {
     if (!options.flat) {
-      var filePath = path.join('src', 'client', 'system-config.ts');
+      var filePath = path.join(this.project.ngConfig.defaults.sourceDir, 'system-config.ts');
```
Comment (hansl) [inline addon/ng2/blueprints/component/index.js]: Could you `path.normalize` this?

File: addon/ng2/blueprints/ng2/index.js [hunk 3]
```diff
@@ -3,23 +3,34 @@ const stringUtils = require('ember-cli-string-utils');
...
+    { name: 'source-dir', type: String, default: 'src', aliases: ['sd'] },
+    { name: 'prefix', type: String, default: 'app', aliases: ['p'] }
+  ],
 
   locals: function(options) {
     //TODO: pull value from config
     this.styleExt = 'css';
     this.version = require(path.resolve(__dirname, '..', '..', '..', '..', 'package.json')).version;
-    
+    const refToTypings = options.sourceDir.split(path.sep).map(() => '..').join('/');
```
Comment (hansl) [inline addon/ng2/blueprints/ng2/index.js]: `path.normalize` this too. Also, instead of splitting by `path.sep`, could you `split(/[\\\/]/g)`?

**Comment by hansl** [comment]
Mostly LGTM, but I think this will fail horribly on Windows because of using `/` in some cases and `path.sep` in others...

**Comment by Brocco** [comment]
`path.sep` is used mostly, the only time I explicitly use `/` is when creating the barrelUrl which is a browser path that systemJS is using so it should be a `/`

As for `path.normalize` I don't see that buying us anything here, and if it is a matter of validating the value of `sourceDir` then that should be done when we read the configuration file, so that it is consistent throughout the application.

File: lib/broccoli/angular2-app.js [hunk 4]
```diff
@@ -23,7 +26,7 @@ class Angular2App extends BroccoliPlugin {
     options = options || {};
 
     this._options = options;
-    this._sourceDir = options.sourceDir || 'src/client';
+    this._sourceDir = ngConfig.defaults.sourceDir || 'src';
```
Comment (hansl) [inline lib/broccoli/angular2-app.js]: Please keep the options if it's passed in. basically `options.sourceDir || ngConfig.defaults.sourceDir`.  The config itself should have a default value. The `|| 'src'` can be kept for this PR, but we need the options for Material (we have 2 apps).

**Comment by hansl** [comment]
LGTM.

**Comment by johnpapa** [comment]
@spboyer Can you take a look at this and run the tests on Windows? @Brocco needs a windows tester. 

```
git fetch origin pull/571/head:pr571
git checkout pr571
npm i
npm test
```

Thanks!

**Comment by Brocco** [comment]
@filipesilva has also dug into the windows build issue and has reported this:

```javascript
import {Component} from 'angular2/core';

@Component({
  moduleId: __moduleName,
  selector: 'testproj2-app',
  templateUrl: 'testproj2.component.html',
  styleUrls: ['testproj2.component.css']
,
  directives: [ROUTER_DIRECTIVES]
})
export class Testproj2App {
}
```
weird comma there
also no `ROUTER_DIRECTIVES` import
also no routeConfig?

The code for injecting those things in can be found in:
> addon/ng2/blueprints/route/index.js

**Comment by hansl** [comment]
I have no idea about the imports. I can't repro it on my computer and it's obviously not the problem with the Appveyor build.

**Pull request merged**